Lib/idlelib/editor.py

```diff
@@ -1281,14 +1281,20 @@ def smart_backspace_event(self, event):
         # Delete whitespace left, until hitting a real char or closest
         # preceding virtual tab stop.
         chars = text.get("insert linestart", "insert")
+        echars = text.get("insert", "insert lineend")
         if chars == '':
             if text.compare("insert", ">", "1.0"):
                 # easy: delete preceding newline
                 text.delete("insert-1c")
             else:
                 text.bell()     # at start of buffer
             return "break"
-        if  chars[-1] not in " \t":
+        if chars.startswith(self.prompt_last_line):
+            isindent = _line_indent_re.fullmatch(chars[len(self.prompt_last_line):])
+        else:
+            isindent = _line_indent_re.fullmatch(chars)
+        istrailing = _line_indent_re.fullmatch(echars)
```

Comment:
Please explain the intent of each user-visible change with both some text and test example with (input text, current 'wrong' result, fixed 'right' result).  Then I can decide first if I agree with 'wrong' and 'right' and only then if I agree that the code change is the best way to get 'right'.  The examples I agree with will be the initial examples needed to test this function. We will also need enough examples of (input, current right result) cover all the branches.

Reply:
I have added 13 tests to hopefully cover the different branches. I think this is resolved but have yet to update the coverage (I will do this once we are happy with the tests).